standalone-metastore/metastore-tools/metastore-benchmarks/src/main/java/org/apache/hadoop/hive/metastore/tools/HMSBenchmarks.java

```diff
@@ -416,15 +416,31 @@ static DescriptiveStatistics benchmarkGetPartitionsByFilter(@NotNull MicroBenchm
     final HMSClient client = data.getClient();
     String dbName = data.dbName;
     String tableName = data.tableName;
+    List<String> partitionList = Arrays.asList("date", "hour");
 
-    BenchmarkUtils.createPartitionedTable(client, dbName, tableName);
+    BenchmarkUtils.createPartitionedTable(client, dbName, tableName, partitionList);
     try {
-      addManyPartitionsNoException(client, dbName, tableName, null,
-              Collections.singletonList("d"), count);
-      return bench.measure(
+      try {
+        Table table = client.getTable(dbName, tableName);
+        List<Partition> partitionValueList = new ArrayList<>();
+        for (int i = 0; i < count / 24; i++) {
```

Comment:
The count would take no effect in some cases, for example if count is less than 24, then no partitions will be added.